File: 0121-best-time-to-buy-and-sell-stock/solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0;
        int cur_min = prices[0];
        int cur_max = prices[0];
        for (int i = 1; i < prices.size(); ++i) {
            profit = max(profit, cur_max - cur_min);
            cur_max = prices[i];
            if (cur_min > prices[i]) {
                cur_min = prices[i];
            }
        }
        profit = max(profit, cur_max - cur_min);
        return profit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minPrice = prices[0]
        maxProfit = 0
        n = len(prices)
        
        for i in range(1, n):
            profit = prices[i] - minPrice
            maxProfit = max(maxProfit, profit)
            minPrice = min(minPrice, prices[i])
        
        return maxProfit
